graphs/c-or-cpp/Prim's-algorithm.c

```diff
@@ -212,4 +214,6 @@ enter the node no of v2 and weight 3
 2
 Output: Cost: 4
 
+Time Complexity:O(ElogV)
```

Comment:
time complexity is write or wrong

Reply:
Depends on the way of **implementation** for the algorithm

For Binary heap yes it is `O(ElogV)`
For Array it is `O(V^2)`
For Fibonacci heap it is `O(E + VlogV)`

Reply:
> Depends on the way of **implementation** for the algorithm
> 
> For Binary heap yes it is `O(ElogV)`
> For Array it is `O(V^2)`
> For Fibonacci heap it is `O(E + VlogV)`

Using priority queue

Reply:
Time complexity will be `O(V^2)` in this particular implementation. Because it is just using linked list to store the values and traversing the entire linked list to find the minimum value in the set of edges ( which is similar to using arrays ) 

@meenagupta5 all the options @TawfikYasser pointed out will use priority queue only, the difference lies in how user is implementing that priority queue. ( a typical implementation uses min-heap for prim's algo )